# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webscheduling/WebTaskSchedulerWorker.h
# Commit: 79b3b0148eb4
# Full Hash: 79b3b0148eb416e0690d3e71723bd68ac67245e7
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-02-27 21:52:52
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Implement scheduler.yield r=jjaschke,dom-core,webidl,smaug
#   
#   Spec: https://wicg.github.io/scheduling-apis/#dom-scheduler-yield
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D234232
# ==============================================================================

diff -r 78181c32bdc6 -r 79b3b0148eb4 dom/webscheduling/WebTaskSchedulerWorker.h
--- a/dom/webscheduling/WebTaskSchedulerWorker.h	Thu Feb 27 16:17:35 2025 +0000
+++ b/dom/webscheduling/WebTaskSchedulerWorker.h	Thu Feb 27 16:17:35 2025 +0000
@@ -20,8 +20,7 @@
 
 class WebTaskWorkerRunnable final : public WorkerSameThreadRunnable {
  public:
-  WebTaskWorkerRunnable(WorkerPrivate* aWorkerPrivate,
-                        WebTaskSchedulerWorker* aSchedulerWorker);
+  explicit WebTaskWorkerRunnable(WebTaskSchedulerWorker* aSchedulerWorker);
 
   MOZ_CAN_RUN_SCRIPT_BOUNDARY
   bool WorkerRun(JSContext* aCx, WorkerPrivate* aWorkerPrivate) override;
@@ -42,8 +41,9 @@
  private:
   ~WebTaskSchedulerWorker() = default;
 
-  nsresult SetTimeoutForDelayedTask(WebTask* aTask, uint64_t aDelay) override;
-  bool DispatchEventLoopRunnable() override;
+  nsresult SetTimeoutForDelayedTask(WebTask* aTask, uint64_t aDelay,
+                                    EventQueuePriority aPriority) override;
+  bool DispatchEventLoopRunnable(EventQueuePriority aPriority) override;
 
   RefPtr<StrongWorkerRef> mWorkerRef;
   bool mWorkerIsShuttingDown{false};